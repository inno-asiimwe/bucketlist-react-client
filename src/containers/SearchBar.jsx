/**
 * Component handles making search quieries
 */
import React, { Component } from 'react';

export default class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = { term: '' };
  }

  /**
   * Function handles typing into the search form
   * @param {string} term - term typed into the search form
   */
  onInputChange(term) {
    this.setState({ term });
    this.props.onSearchTermChange(term);
  }
  render() {
    return (
      <form className="form-inline my-2 my-lg-0">
        <input
          className="form-control mr-sm-2"
          type="text"
          placeholder="Search Bucketlists"
          value={this.state.term}
          onChange={event => this.onInputChange(event.target.value)}
        />
      </form>
    );
  }
}

