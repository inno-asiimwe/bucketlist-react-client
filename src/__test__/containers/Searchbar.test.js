import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SearchBar from '../../containers/SearchBar';

describe('SearchBar component', () => {
  it('Matches its snapshot', () => {
    const output = shallow(<SearchBar />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
