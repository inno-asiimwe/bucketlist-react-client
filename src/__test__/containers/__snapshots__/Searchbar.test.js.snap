// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component Matches its snapshot 1`] = `
<form
  className="form-inline my-2 my-lg-0"
>
  <input
    className="form-control mr-sm-2"
    onChange={[Function]}
    placeholder="Search Bucketlists"
    type="text"
    value=""
  />
</form>
`;
